src/include/self_driving/forecast/workload_forecast.h

```diff
@@ -42,7 +53,7 @@ class WorkloadForecast {
   uint64_t num_sample_{5};
 
   std::vector<WorkloadForecastSegment> forecast_segments_;
-  uint64_t num_forecast_segment_{0};
+  uint64_t num_forecast_segment_;
   uint64_t forecast_interval_;
   uint64_t optimizer_timeout_{10000000};
```

Comment:
Our convention is to use Getters and Setters for **all** member variable access unless it's a super simple data struct. This can give us control on whether the member is changeable, whether there should be any validation check before updating a member, etc. And for read-only access, we generally use a Getter with **const reference**. Could you refactor the access to all the members in `WorkloadForecast` with Getters?